.github/workflows/release.yml

```diff
@@ -0,0 +1,76 @@
+name: Build and publish distribution
+
+on:
+  release:
+    types: [ published ]
+
+jobs:
+  publish-test:
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v3
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+      - name: Install build dependencies
+        run: pip install build
+      - name: Build distributions
+        run: python -m build
+      - name: Publish package to PyPI
+        if: github.repository == 'cleanlab/cleanvision' && github.event_name == 'release' && github.event.action == 'published'
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+          repository-url: https://test.pypi.org/legacy/
+  publish:
+    needs: [ publish-test ]
+    # This condition ensures publish-test succeeded before running this job
+    if: needs.publish-test.result == 'success'
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v3
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+      - name: Install build dependencies
+        run: pip install build
+      - name: Build distributions
+        run: python -m build
+      - name: Publish package to PyPI
+        if: github.repository == 'cleanlab/cleanvision' && github.event_name == 'release' && github.event.action == 'published'
```

Comment:
This is no longer necessary, right?

You have this single trigger event for the workflow
```yml
on:
  release:
    types: [ published ]
```


along with the `if: needs.PREVIOUS_JOB.result == 'success'` check.